### ISSUE TITLE
stableSort sorts the whole array

### ISSUE THREAD

**Issue by bal-a** [scala/bug#12568]
## Reproduction steps

Scala version: 2.13.8

```
scala> import scala.util.Sorting.stableSort
import scala.util.Sorting.stableSort

scala> val unsortedArray = Array(99,1,3,2)
val unsortedArray: Array[Int] = Array(99, 1, 3, 2)

scala> stableSort(unsortedArray, 1, 3)

scala> unsortedArray
val res4: Array[Int] = Array(1, 2, 3, 99)
```

## Problem

The signature of the method `stableSort` implies that you can sort a slice, but the `from` and `until` arguments are ignored in here: https://github.com/scala/scala/blob/4b2151cbe4c857d49556b70c3fa0f5e236fd7754/src/library/scala/util/Sorting.scala#L241

I think that the issue can be fixed just by passing both arguments `from: Int, until: Int` to java sorting methods.

**Comment by SethTisue** [comment]
volunteer from @scala/collections perhaps?

**Comment by abalys** [comment]
> volunteer from @scala/collections perhaps?
> 
> @bal-a or are you interested in PRing a fix yourself?

I'll create a PR :))

**Comment by bal-a** [comment]
> volunteer from @scala/collections perhaps?
> 
> @bal-a or are you interested in PRing a fix yourself?

I'll create a PR :)

**Comment by som-snytt** [comment]
I looked at it last night and assumed I was too tired for the API to make sense to me.

The ranged API was added to fix `ArrayBuffer#sortInPlace`, where some of the array is unoccupied.

As described in the Scaladoc, the convenience API is to let you choose between quick and merge sorts. The Javadoc makes the distinction in the implementation note. This class chooses based on ordering supplied, but the Scaladoc also implies maybe you care about memory usage.

Besides ordering, supplying a range could imply "I want a merge sort."

OTOH, just supplying the range on the Java quicksort flavors is probably benign.

This class is opinionated about the float case.
